src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -17,21 +17,11 @@ public StorageImpl() {
     @Override
     public void put(K key, V value) {
         boolean isExist = false;
-        if (key == null) {
-            for (int i = 0; i < index; i++) {
-                if (key == keys[i]) {
-                    keys[i] = key;
-                    values[i] = value;
-                    isExist = true;
-                }
-            }
-        } else {
-            for (int i = 0; i < index; i++) {
-                if (key.equals(keys[i])) {
-                    keys[i] = key;
-                    values[i] = value;
-                    isExist = true;
-                }
+        for (int i = 0; i < index; i++) {
+            if ((key == null && key == keys[i]) || (key != null && key.equals(keys[i]))) {
+                keys[i] = key;
+                values[i] = value;
+                isExist = true;
```

Comment:
```suggestion
               return;
```
тоді можна не використовувати змінну isExist і перевірку if нижче